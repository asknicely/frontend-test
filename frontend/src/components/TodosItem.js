import React from 'react'
import { Link } from "react-router-dom"

const Item = (props) => {
    return (
        <tr>
            <td>{props.id}</td>
            <td>{props.user}</td>
            <td>{!props.link ? props.description : <Link to={`/todo/${props.id}`}>{props.description}</Link>}</td>
            <td>{parseInt(props.completed) === 1 ? 'Completed' : ''}</td>
            <td nowrap="true">

                <button 
                    type="button" 
                    className="btn btn-xs btn-danger custom-button"
                    onClick={() => { props.handleDelete(props.id) }}
                >
                <span className="glyphicon glyphicon-remove glyphicon-white"></span>
                </button>

                <button 
                    type="button" 
                    className="btn btn-xs btn-success custom-button"
                    onClick={() => { props.handleComplete(props.id) }}
                    style={{ display: parseInt(props.completed) === 1 ? 'none' : 'inline-block' }}
                >
                <span className="glyphicon glyphicon-ok-sign glyphicon-white"></span>
                </button>

            </td>
        </tr>
    )
}

export default Item