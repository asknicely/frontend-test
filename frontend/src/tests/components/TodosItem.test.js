import React, { useReducer } from 'react';
import { shallow } from 'enzyme';
import TodosItem from '../../components/TodosItem';
import TodosFixtures from '../fixtures/todos'

test('should render TodosItem correctly :: 2nd Item', () => {

	const item = 2
	const wrapper = shallow(
		<TodosItem 
			key={TodosFixtures[item].id}
			id={TodosFixtures[item].id}
			user={TodosFixtures[item].user_id}
			description={TodosFixtures[item].description}
			completed={TodosFixtures[item].completed}
			handleDelete={TodosFixtures[item].handleDelete}
			handleComplete={TodosFixtures[item].handleComplete}
			link={true}			
		/>
	);

	expect(wrapper).toMatchSnapshot();
	
});
