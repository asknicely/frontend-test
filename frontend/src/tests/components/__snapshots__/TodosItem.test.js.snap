// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render TodosItem correctly :: 2nd Item 1`] = `
<tr>
  <td>
    3
  </td>
  <td>
    345
  </td>
  <td>
    <Link
      to="/todo/3"
    >
      Going out to university
    </Link>
  </td>
  <td />
  <td
    nowrap="true"
  >
    <button
      className="btn btn-xs btn-danger custom-button"
      onClick={[Function]}
      type="button"
    >
      <span
        className="glyphicon glyphicon-remove glyphicon-white"
      />
    </button>
    <button
      className="btn btn-xs btn-success custom-button"
      onClick={[Function]}
      style={
        Object {
          "display": "inline-block",
        }
      }
      type="button"
    >
      <span
        className="glyphicon glyphicon-ok-sign glyphicon-white"
      />
    </button>
  </td>
</tr>
`;
